ParquetClassLibrary/Sandbox/Parquets/ParquetParent.cs

```diff
@@ -1,22 +1,196 @@
+﻿using System;
+using System.Collections.Generic;
+using Newtonsoft.Json;
+using ParquetClassLibrary.Sandbox.ID;
+using ParquetClassLibrary.Utilities;
+
 namespace ParquetClassLibrary.Sandbox.Parquets
 {
     /// <summary>
-    /// Parent for a all sandbox-mode parquets.
+    /// Models a sandbox-mode parquet.
     /// </summary>
-    public abstract class ParquetParent
+    public abstract class ParquetParent : IEquatable<ParquetParent>
     {
+        /// <summary>Used to ensure only unique IDs are used during parquet creation.</summary>
+        private static readonly HashSet<ParquetID> KnownIDs = new HashSet<ParquetID>();
```

Comment:
This should be in all Parquets, and actually is (sort of) already. The _table variable has a list of Keys that already does this.

Reply:
Interesting, okay.  I had to think about this one for a bit but I will follow your advice.